### DISCUSSION TITLE
Ditto Sink connection issue

### DISCUSSION THREAD

**Discussion by mihau1211** [apache/streampipes#2342] [Q&A] [answered]
Hi, I'm encountering an issue while trying to connect Ditto to Ditto Sink in StreamPipes. I have a Ditto instance running on Docker, and I've verified the connection using wscat with basic Authorization. I've set the credentials as "ditto:ditto" and added the hashed value of "ditto" to nginx.htpasswd.

From the Ditto side, everything appears to be working fine, I can use the API via HTTP, and wscat returns a "connected" message.
![image](https://github.com/apache/streampipes/assets/73185784/ddae6b3b-905b-4ba9-ac34-4963577d6dad)

On the StreamPipes UI, I'm encountering the following error:
![image](https://github.com/apache/streampipes/assets/73185784/bbeee9f7-b6cf-443a-92e5-67f47b3442fe)

Ditto Sink Configuration:
![image](https://github.com/apache/streampipes/assets/73185784/937f1a71-7503-4588-b9df-17107e8825b5)

Tbh, I'm running out of ideas, not sure if it is the issue related to the Ditto instance or StreamPipes. I will be grateful for any help.

**Comment by obermeier** [comment, marked as answer]
Hi @mihau1211, 
it possibly helps if you replace localhost in the Dito API endpoint configuration with the ip address of your hostmachine.
wss://{ ip of your host }:8080/ws/2

**Comment by bossenti** [comment]
Hi @mihau1211,

I assume @obermeier's response has solved your problem - if not feel free to open the discussion or comment again
